### PR TITLE
Sanitize json configuration

### PR DESCRIPTION
**Pull request by wallaceSF**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| **Issue**         | https://mundipagg.atlassian.net/browse/CONECT-1104
| **What?**         | Does sanitize JSON when it comes with special characters
| **Why?**          | because the JSON broken

### CODE REVIEW

**Comment by wallaceSF** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ src/Kernel/Repositories/ConfigurationRepository.php  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/mundipagg/ecommerce-module-core/pullRequest?prid=5891213&bid=19288428)

**Review by GabrielDeveloper — APPROVED** [review]

File: src/Kernel/Repositories/ConfigurationRepository.php [hunk 1]
```diff
@@ -13,6 +13,15 @@ class ConfigurationRepository extends AbstractRepository
     protected function create(AbstractEntity &$object)
     {
         $jsonEncoded = json_encode($object);
+
+        $jsonEncoded = trim(
+            preg_replace(
+                '/\\\s+\\\s\\\r\\\n|\\\r|\\\n\\\r|\\\n/m',
```
Comment (michelpl) [inline src/Kernel/Repositories/ConfigurationRepository.php]: Coloca esse regex numa variável

File: src/Kernel/Repositories/ConfigurationRepository.php [hunk 2]
```diff
@@ -36,6 +45,15 @@ protected function update(AbstractEntity &$object)
         }
 
         $jsonEncoded = json_encode($object);
+
+        $jsonEncoded = trim(
```
Comment (michelpl) [inline src/Kernel/Repositories/ConfigurationRepository.php]: Código duplicado pode ser extraído para um outro método

**Review by michelpl — CHANGES_REQUESTED** [review]

File: src/Kernel/Repositories/ConfigurationRepository.php [hunk 3]
```diff
@@ -10,9 +10,28 @@
 
 class ConfigurationRepository extends AbstractRepository
 {
+    private $pattern = '/\\\s+\\\s\\\r\\\n|\\\r|\\\n\\\r|\\\n/m';
+
+    /**
+     * @param string $jsonEncoded
+     * @return string
+     */
+    private function treatJsonString($jsonEncoded)
```
Comment (michelpl) [inline src/Kernel/Repositories/ConfigurationRepository.php]: O método pode se chamar removeSpecialCharacters

**Review by michelpl — CHANGES_REQUESTED** [review]

**Pull request merged**